src/entity.cpp

```diff
@@ -1693,26 +1693,54 @@ void Entity::handleEffects(Stat* myStats)
 
 	// healing over time
 	int healring = 0;
+	int healthRegenInterval = 0;
 	if ( myStats->ring != NULL )
 	{
 		if ( myStats->ring->type == RING_REGENERATION )
 		{
 			if ( myStats->ring->beatitude >= 0 )
 			{
-				healring = 1;
+				healring++;
```

Comment:
For the record, we prefer pre-increment to post-increment, unless the behavior of post-increment is what is semantically desired.